inc/src/Services/Config/CoreConfigLoader.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace MyBB\Services\Config;
+
+use Illuminate\Contracts\Support\Arrayable;
+use MyBB\Contracts\Config\LoaderInterface;
+
+class CoreConfigLoader implements LoaderInterface
+{
+
+    /**
+     * @var array|null config
+     */
+    private $config;
+
+    /**
+     * @var string path
+     */
+    private $path;
+
+    /**
+     * CoreConfigLoader constructor.
+     * @param $path
+     */
+    public function __construct($path)
+    {
+        $this->path = $path;
+        $this->config = null;
+    }
+
+    /**
+     * Retrieve the configuration file
+     */
+    private function loadConfig()
+    {
+        require $this->path;
```

Comment:
Should validate the path exists, or else this could result in a PHP error. Should also declare `$config = array()` first.